Add change log for v10.0.0-rc.9 (#658)
mapbox/mapbox-maps-android: CHANGELOG.md

Mapbox welcomes participation and contributions from everyone.

# 10.0.0-rc.9 September 22, 2021

**The Mapbox Maps SDK for Android has moved to release candidate status and is now ready for production use.**

## Features ✨ and improvements 🏁

> * Fix documentation for OnMapIdleListener and CameraChangeListeners. ([#645](https://github.com/mapbox/mapbox-maps-android/pull/645))

Reviewer: ```suggestion
* Fix documentation for `OnMapIdleListener` and `CameraChangeListeners`. ([#645](https://github.com/mapbox/mapbox-maps-android/pull/645))
```

> * [docs] Fix documentation for OnMapIdleListener and CameraChangeListeners. ([#645](https://github.com/mapbox/mapbox-maps-android/pull/645))

Reviewer 2: remove `[docs]`?

> * Add support for SymbolZOrder in PointAnnotationManager ([#638](https://github.com/mapbox/mapbox-maps-android/pull/638))

Reviewer 2: ```suggestion
* Add support for `SymbolZOrder` property in PointAnnotationManager. ([#638](https://github.com/mapbox/mapbox-maps-android/pull/638))
```

> * Add support for PromoteId to be used with Feature State API. ([#636](https://github.com/mapbox/mapbox-maps-android/pull/636))

Reviewer: ```suggestion
* Add support for `PromoteId` to be used with Feature State API. ([#636](https://github.com/mapbox/mapbox-maps-android/pull/636))
```

> * Add support for PromoteId to be used with Feature state API. ([#636](https://github.com/mapbox/mapbox-maps-android/pull/636))

Reviewer 2: ```suggestion
* Add support for PromoteId to be used with Feature State API. ([#636](https://github.com/mapbox/mapbox-maps-android/pull/636))
```

Reviewer 2: maybe, imho just looks better 😄

> * Expose optimizeForTerrain flag ([#654](https://github.com/mapbox/mapbox-maps-android/pull/654))

Reviewer 2: ```suggestion
* Expose `optimizeForTerrain` flag that could be applied to the `MapView` in xml. ([#654](https://github.com/mapbox/mapbox-maps-android/pull/654))
```

> * Enable instant transitions for data driven symbol layer properties.

Reviewer 2: Add link to PR where gl-native bump did happen?

## Bug fixes 🐞

> * onStyleLoaded/onMapLoaded callbacks are called even if hosting fragment/activity is stopped ([#629](https://github.com/mapbox/mapbox-maps-android/pull/629))

Reviewer 2: ```suggestion
* OnStyleLoaded / OnMapLoaded callbacks are invoked even if hosting fragment/activity is in stopped state. ([#629](https://github.com/mapbox/mapbox-maps-android/pull/629))
```

> * OnStyleLoaded / OnMapLoaded callbacks are invoked even if hosting fragment/activity is in stopped state. ([#629](https://github.com/mapbox/mapbox-maps-android/pull/629))

Reviewer: ```suggestion
* `OnStyleLoaded` / `OnMapLoaded` callbacks are invoked even if hosting fragment/activity is in stopped state. ([#629](https://github.com/mapbox/mapbox-maps-android/pull/629))
```

> * Fix drag annotation blink issue ([#639](https://github.com/mapbox/mapbox-maps-android/pull/639))

Reviewer 2: ```suggestion
* Fix drag annotation blink when drag ends. ([#639](https://github.com/mapbox/mapbox-maps-android/pull/639))
```

> * Apply annotation manager properties to drag layer to keep annotations the same while dragging ([#640](https://github.com/mapbox/mapbox-maps-android/pull/640))

Reviewer 2: ```suggestion
* Apply annotation manager properties to the drag layer to keep annotations the same while dragging. ([#640](https://github.com/mapbox/mapbox-maps-android/pull/640))
```

Reviewer 2: @Chaoba was does `the same` means here? same comparing to what?

Reviewer: Annotation properties like text color will change to default while dragging.

> * Fix point annotation update bitmap affect others issue. ([#633](https://github.com/mapbox/mapbox-maps-android/pull/633))

Reviewer 2: ```suggestion
* Fix point annotation updating all same content bitmaps instead of one particular. ([#633](https://github.com/mapbox/mapbox-maps-android/pull/633))
```

> * Fix getStyle returns null after adding a source ([#643](https://github.com/mapbox/mapbox-maps-android/pull/643))

Reviewer 2: ```suggestion
* Fix `MapboxMap#getStyle` returning null after adding a new source when style was loaded before. ([#643](https://github.com/mapbox/mapbox-maps-android/pull/643))
```

> * Enable annotations to set null value to nullable properties. ([#650](https://github.com/mapbox/mapbox-maps-android/pull/650))

Reviewer 2: ```suggestion
* Allow setting null explicitly to annotation nullable properties. ([#650](https://github.com/mapbox/mapbox-maps-android/pull/650))
```

> * Fix set camera crash bug. ([#652](https://github.com/mapbox/mapbox-maps-android/pull/652))

Reviewer 2: ```suggestion
* Fix `std::exception` happing rarely when `MapboxMap#setCamera()` is called inside animation plugin. ([#652](https://github.com/mapbox/mapbox-maps-android/pull/652))
```

* Fix transition between layers with all constant properties.
* Fix rendering artifact for a line layer, when its `line-gradient` property is set at runtime.
* Don't draw SDF images in `text-field` and issue warning for it.

> * Fix incorrect return from StyleManager#getStyleLayerPropertyDefaultValue for `text-field`, now the default value is set to `["format", "" , {}]`.

Reviewer 2: Add ref to gl-native pr
